### ISSUE TITLE
Porting FDL to Hadoop - MapReduce framework for C/C++ 

### ISSUE THREAD

**Issue by unixneo** [godlikemouse/ForceDirectedLayout#4]
Hey godlikemouse!

I've noticed that processing a FDL graph of around 25,000 nodes edges takes around 40 (plus or minus depending on server load) minutes on an 8 core Linux server with 32 GB of RAM. 

We are considering adding hooks to this code to your FDL code:

MapReduce framework for C/C++ open-sourced by Google

http://www.developer-tech.com/news/2015/feb/19/mapreduce-framework-cc-open-sourced-google/

Based on your extensive experience, do you think that it is feasible to increase the performance of FDL by modifying FDL to run as native MapReduce C++ code for Hadoop?

Or, is the FDL building process, based on your experience, not conducive for distributed processing, as I suspect?

**Comment by unixneo** [comment]
Dear godlikemouse,

We found answers online here:

https://dalspace.library.dal.ca/bitstream/handle/10222/65249/Lapierre-Nathan-MCSc-Dec-2015.pdf

and other research papers;

so please close this issue.

Thanks.

**Comment by godlikemouse** [comment]
Hi unixneo,

I just wanted to follow up with you regarding this issue.  In all honesty, what I've created was really more of a quick and dirty approach to handling FDL.  If you're going to go production with this, then you're probably going to want to grab a more commercial FDL system optimized for speed.  In my line of work we do FDL on the fly and use commercial grade algorithms that are trimmed and optimized to handle large topologies.  This system I've created will work well for small systems, but large topologies are going to be slow and possibly problematic.  Hope this helps.
